saul-examples/src/main/java/edu/illinois/cs/cogcomp/saulexamples/data/CLEFImageReader.java

```diff
@@ -326,21 +350,38 @@ private void getTrainingImages() throws IOException {
     // Loading Testing Images
     /*******************************************************/
     private void getTestImages() throws IOException {
-        String testImage = path + "/testing.mat";
-        File d = new File(testImage);
+        if (readFullData) {
+            String testImage = path + "/testing.mat";
+            File d = new File(testImage);
 
-        if (!d.exists()) {
-            throw new IOException(testImage + " does not exist!");
-        }
-        MatFileReader matTrainreader = new MatFileReader(testImage);
+            if (!d.exists()) {
+                throw new IOException(testImage + " does not exist!");
+            }
+            MatFileReader matTrainreader = new MatFileReader(testImage);
 
-        double[][] testing = ((MLDouble) matTrainreader.getMLArray("testing")).getArray();
+            double[][] testing = ((MLDouble) matTrainreader.getMLArray("testing")).getArray();
 
-        if (testing.length > 1) {
-            for (int i = 0; i < testing.length; i++)
-            {
-                int imageID = (int)testing[i][0];
-                testData.add(Integer.toString(imageID));
+            if (testing.length > 1) {
+                for (int i = 0; i < testing.length; i++) {
+                    int imageID = (int) testing[i][0];
+                    testData.add(Integer.toString(imageID));
+                }
+            }
+        } else {
+            String testImage = path + "/sprl2017_gold.xml";
+            File f = new File(testImage);
+
+            if (!f.exists()) {
+                throw new IOException(testImage + " does not exist!");
+            }
+            NlpXmlReader reader = new NlpXmlReader(testImage, "SCENE", "SENTENCE", null, null);
+            List<Document> documentList = reader.getDocuments();
+
+            for (Document d : documentList) {
+                String name = d.getPropertyFirstValue("IMAGE");
+                String s = name.substring(name.lastIndexOf("/") + 1);
+                String[] label = s.split("\\.");
+                testData.add(label[0]);
```

Comment:
Both of `getTrainingImages` and `getTestImages` are almost identical and there are many duplicate lines of code, why not refactor common functions from those parts? or maybe combine the two and add a parameter to determine test and train?
[Duplicate code is bad. ](https://sourcemaking.com/refactoring/smells/duplicate-code)